### PR TITLE
Retail recommend

### PR DESCRIPTION
**Pull request by aarsanjani**
*Issue #, if available:* retail recommendation using factorization machines on sagemaker 

*Description of changes:*
retail recommendation using factorization machines on sagemaker 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: use-cases/retail_recommend/deploy.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+parser.add_argument('--endpoint-name', type=str)
+args = parser.parse_args()
+
+region = args.region
+boto3.setup_default_session(region_name=region)
+sagemaker_boto_client = boto3.client('sagemaker')
+
+#name truncated per sagameker length requirements (63 char max)
+endpoint_config_name=f'{args.model_name[:56]}-config'
+existing_configs = sagemaker_boto_client.list_endpoint_configs(NameContains=endpoint_config_name)['EndpointConfigs']
```
[the remark this answers is not in the document]
Comment (aarsanjani) [reply use-cases/retail_recommend/deploy.py:20]: Does not pertain to this code.

File: use-cases/retail_recommend/deploy.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+if not existing_endpoints:
+    create_endpoint_response = sagemaker_boto_client.create_endpoint(
+        EndpointName=args.endpoint_name,
+        EndpointConfigName=endpoint_config_name)
+
+endpoint_info = sagemaker_boto_client.describe_endpoint(EndpointName=args.endpoint_name)
+endpoint_status = endpoint_info['EndpointStatus']
+
+while endpoint_status == 'Creating':
+    endpoint_info = sagemaker_boto_client.describe_endpoint(EndpointName=args.endpoint_name)
```
[the remark this answers is not in the document]
Comment (aarsanjani) [reply use-cases/retail_recommend/deploy.py:45]: Does not pertain to this code.

File: use-cases/retail_recommend/deploy.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        ProductionVariants=[{
+            'InstanceType': args.endpoint_instance_type,
+            'InitialVariantWeight': 1,
+            'InitialInstanceCount': 1,
+            'ModelName': args.model_name,
+            'VariantName': 'AllTraffic'
+        }]
+    )
+
+existing_endpoints = sagemaker_boto_client.list_endpoints(NameContains=args.endpoint_name)['Endpoints']
```
[the remark this answers is not in the document]
Comment (aarsanjani) [reply use-cases/retail_recommend/deploy.py:34]: Does not pertain to this code.

**Comment by mchoi8739** [comment]
This notebook series would fail the CI due to StoreMagic. Reviewed offline. Approving and merging.

**Review by mchoi8739 — APPROVED** [review]

**Pull request merged**